pandas/core/groupby/generic.py

```diff
@@ -1094,13 +1111,7 @@ def cast_agg_result(
             # see if we can cast the values to the desired dtype
             # this may not be the original dtype
 
-            if isinstance(values, Categorical) and isinstance(result, np.ndarray):
-                # If the Categorical op didn't raise, it is dtype-preserving
-                # We get here with how="first", "last", "min", "max"
-                result = type(values)._from_sequence(result.ravel(), dtype=values.dtype)
-                # Note this will have result.dtype == dtype from above
-
-            elif (
```

Comment:
in conjunction with #41086, cast_agg_result becomes a 1-liner and can be cleanly tacked onto the end of `py_fallback`